RUFAS/routines/manure/manure_treatments/anaerobic_digestion.py

```diff
@@ -74,40 +77,61 @@ def _calc_anaerobic_digestion_daily_output(
+        )
+        total_methane_generation_mass = total_methane_generation_volume * GasEmissionConstants.AD_METHANE_DENSITY
+        AD_carbon_dioxide = (
+            total_methane_generation_volume * GasEmissionConstants.AD_CARBON_DIOXIDE_TO_METHANE_RATIO
+        ) * GasEmissionConstants.AD_CARBON_DIOXIDE_DENSITY
+        AD_VS_destruction = total_methane_generation_mass + AD_carbon_dioxide
+
+        new_daily_output.liquid_manure_total_solids = (
+            self._current_manure_treatment_daily_input.liquid_manure_total_solids - AD_VS_destruction
+        )
+        new_daily_output.liquid_manure_total_volatile_solids = (
+            self._current_manure_treatment_daily_input.liquid_manure_total_volatile_solids - AD_VS_destruction
+        )
         new_daily_output.liquid_manure_total_degradable_volatile_solids = (
             self._current_manure_treatment_daily_input.liquid_manure_total_degradable_volatile_solids
-            - (new_daily_output.biogas * GasEmissionConstants.AD_METHANE_TO_METHANE_CARBON_DIOXIDE_RATIO)
+            - AD_VS_destruction
         )
         new_daily_output.liquid_manure_total_non_degradable_volatile_solids = (
             self._current_manure_treatment_daily_input.liquid_manure_total_non_degradable_volatile_solids
         )
-        new_daily_output.liquid_manure_total_volatile_solids = (
-            new_daily_output.liquid_manure_total_degradable_volatile_solids
-            + new_daily_output.liquid_manure_total_non_degradable_volatile_solids
+
+        new_daily_output.daily_final_manure_volume = (
+            self._current_manure_treatment_daily_input.liquid_manure_daily_volume
+            - (AD_VS_destruction / ManureConstants.SLURRY_MANURE_DENSITY)
         )
-        new_daily_output.liquid_manure_total_solids = (
-            self._current_manure_treatment_daily_input.liquid_manure_total_solids
-            - (new_daily_output.biogas * GasEmissionConstants.AD_METHANE_TO_METHANE_CARBON_DIOXIDE_RATIO)
+        methane_leakage = GasEmissionsCalculator.calculate_digester_methane_leakage(
+            total_methane_generation_mass, self.config.digester_methane_leakage_fraction
+        )
+        captured_methane_generation_volume = total_methane_generation_volume - (
+            methane_leakage / GasEmissionConstants.AD_METHANE_DENSITY
+        )
+        captured_methane_generation_mass = total_methane_generation_mass - methane_leakage
+        captured_methane_energy_content = GasEmissionsCalculator.calculate_methane_energy_content(
+            methane_mass=captured_methane_generation_mass
         )
         new_daily_output.heating_input_energy = heating_input_energy
         new_daily_output.evaporated_water = self.config.evaporation_fraction * daily_final_manure_volume
-        new_daily_output.biogas_energy_content = biogas_energy_content
+        new_daily_output.methane_energy_content = captured_methane_energy_content
```

Comment:
```suggestion
        new_daily_output.biogas_energy_content = captured_methane_energy_content
```
Reverting output name change but preserve new internal calculation of biogas methane content